WestLaw UK.js

```diff
@@ -0,0 +1,163 @@
+		return "statute";
+	}
+	else if (doc.title.includes("s.")) {
+		return "statuteSection";
+	}
+	else {
+		return false;
+	}
+}
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "case") {
+		return scrapeCase(doc, url);
+	}
+	else if (detectWeb(doc, url) == "statute") {
+		return scrapeStatute(doc, url);
+	}
+	else if (detectWeb(doc, url) == "statuteSection") {
+		return scrapeStatuteSection(doc, url);
+	}
+	else {
+		return false;
+	}
+}
+
+
+function scrapeCase(doc, url) {
+	var translator = Zotero.loadTranslator('web');
+	// Embedded Metadata
+	translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');
+	// translator.setDocument(doc);
+	
+	translator.setHandler('itemDone', function (obj, item) {
+		// TODO adjust if needed:
+		
+		var citationArray = parseCitationList(doc.getElementsByClassName("co_docContentMetaField")[3].innerText);
+		
+		item.reporter = citationArray[3];
+		item.reporterVolume = citationArray[2];
+		item.firstPage = citationArray[4];
+		item.dateDecided = citationArray[1];
+		item.court = /Court\n(.+)/.exec(doc.getElementById("co_docContentCourt").innerText)[1];
+		item.abstractNote = "";
+		item.complete();
+		
+		
+	});
+
+	translator.getTranslatorObject(function (trans) {
+		trans.itemType = "case";
+		// TODO map additional meta tags here, or delete completely
+
+		trans.doWeb(doc, url);
+	});
+}
+
+function scrapeStatute(doc, url) {
+	var translator = Zotero.loadTranslator('web');
+	// Embedded Metadata
+	translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');
+	// translator.setDocument(doc);
+	
+	translator.setHandler('itemDone', function (obj, item) {
+		// TODO adjust if needed:
+		var fullTitle = doc.getElementById("co_documentTitle").innerText;
+		item.date = /(\D+) (\d+)/.exec(fullTitle)[2];
+		item.title = /(\D+) (\d+)/.exec(fullTitle)[1];
+		item.abstractNote = "";
+		item.complete();
+	});
+
+	translator.getTranslatorObject(function (trans) {
+		trans.itemType = "statute";
+		// TODO map additional meta tags here, or delete completely
+
+		trans.doWeb(doc, url);
+	});
+}
+
+function scrapeStatuteSection(doc, url) {
+	var translator = Zotero.loadTranslator('web');
+	// Embedded Metadata
+	translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');
+	// translator.setDocument(doc);
+	
+	translator.setHandler('itemDone', function (obj, item) {
+		// TODO adjust if needed:
+		var fullTitle = /([^\n]+)/.exec(doc.getElementsByClassName("co_title noTOC")[0].innerText)[0];
+		item.date = /(\D+) (\d+)/.exec(fullTitle)[2];
+		item.title = /(\D+) (\d+)/.exec(fullTitle)[1];
+		item.abstractNote = "";
+		item.complete();
+	});
+
+	translator.getTranslatorObject(function (trans) {
+		trans.itemType = "statute";
+		// TODO map additional meta tags here, or delete completely
+
+		trans.doWeb(doc, url);
+	});
+}
+
+
+function parseCitationList(citList){
+	var allCitations = citList.match(/^[\[|\(](\d+)[\]|\)] (\d*) ?([A-z\. ]+) ([\w]+)$/gm); //annoyingly matchAll is not supported
```

Comment:
Will it always always take this form? Feels like there *must* be a better way. Check if the page has any relevant metadata in `<meta>` tags or hidden fields?

Reply:
I hate this as much as you do, I promise. Luckily law case citations take a pretty standard form, but there really is no way to get this from the page 😓